simulator/ns-3.35/src/point-to-point/model/switch-mmu.cc

```diff
@@ -283,20 +311,24 @@ bool SwitchMmu::CheckIngressAdmission(uint32_t port, uint32_t qIndex, uint32_t p
 
 	switch (type) {
 	case LOSSY:
-		if ( (psize + ingress_bytes[port][qIndex] > Threshold(port, qIndex, "ingress", type) && psize + ingress_bytes[port][qIndex] > reserveIngress[port][qIndex])
-		        // || (psize + ingressPoolUsed > ingressPool )
-		        || (psize + totalSharedUsed > bufferPool) ) {
+		if ( (psize + ingress_bytes[port][qIndex] > Threshold(port, qIndex, "ingress", type)
+		        && psize + ingress_bytes[port][qIndex] > reserveIngress[port][qIndex]) // if ingress bytes is greater than the ingress threshold AND if the reserved is usedup
+		        || (psize + (totalUsed-xoffTotalUsed) > ingressPool) // if the ingress pool is full. With DT, this condition is redundant. This is just to account for any badly configured buffer or buffer sharing if any.
+		        || (psize + totalSharedUsed > bufferPool) )  // or if the shared buffer is full
+		{
 			return false;
 		}
 		else {
 			return true;
 		}
 		break;
 	case LOSSLESS:
-		if ( ( (psize + ingress_bytes[port][qIndex] > reserveIngress[port][qIndex])
-		        && (psize + ingress_bytes[port][qIndex] > Threshold(port, qIndex, "ingress", type) + xoff[port][qIndex]) )
-		        || (psize + ingressPoolUsed > ingressPool + xoffTotal)
-		        || (psize + totalSharedUsed > bufferPool)  ) {
+		if ( ( (psize + ingress_bytes[port][qIndex] > reserveIngress[port][qIndex]) // if reserved is used up
+		        && (psize + GetHdrmBytes(port,qIndex) > xoff[port][qIndex]) && GetHdrmBytes(port,qIndex)>0 ) // and if per queue headroom is used up.
+		        || (xoffTotalUsed >= xoffTotal ) // or if the headroom pool is full
+		        || (psize + totalUsed > ingressPool + xoffTotal) // if the ingresspool+headroom is full. With DT, this condition is redundant. This is just to account for any badly configured buffer or buffer sharing if any.
+		        || (psize + totalSharedUsed > bufferPool)  ) // if the shared buffer is full
```

Comment:
Again, it seems that totalSharedUsed = totalUsed

Reply:
It seems that comment "the shared buffer is full" is inaccurate.  You can say "The switch buffer is physically used up".